### ISSUE TITLE
1-Click-Upgrade 1.7.6.0 -> 1.7.6.1: Module folders contain composer related files

### ISSUE THREAD

**Issue by lichesser** [PrestaShop/PrestaShop#15668]
Hello,

I just did a PS 1.7.6.0 to 1.7.6.1 upgrade with the 1-click-upgrader.

Afterwards almost all module folders contained a new `composer.json` and `composer.lock` file and a `vendor` and `vendor/composer` folder with the same PHP files.
Some module folders also contain a new `Readme.md` file .

Is this expected? As those files are not present in the release zip file?
Can these files be deleted after the upgrade?

**Comment by khouloudbelguith** [comment]
Hi @lichesser,

Thanks for your report.
What is the exact version of the module 1-click upgrade did you use?
What are those exact modules?
after the upgrade of your shop, those modules are upgraded too?

Thanks!

**Comment by lichesser** [comment]
Hello,

- What is the exact version of the module 1-click upgrade did you use?
  - 1-Click Upgrade (`autoupgrade`) module is version: 4.9.0

- What are those exact modules?
  - The list of modules that now have the above mentioned folders and files is too long to reproduce here. Of all modules installed ~70 it seems 65 have those files/folders now.

- after the upgrade of your shop, those modules are upgraded too?
  - No. It seems only the following modules were really upgraded. I think those were expected, but I don't exactly remember what the 1-click-upgrader predicted what would be changed. 
    - `ps_emailsubscription` (2.4.0 -> 2.5.0)
    - `ps_facetedsearch` (3.2.0 -> 3.3.0)
    - `psaddonsconnect` (2.0.0 -> 2.1.0)

I noticed another strange thing after the upgrade.

The `<version>` tag in `config.xml` for `ps_linklist` was changed from `3.0.4` to `3.0.3` but the actually installed code files are definitely version `3.0.4` (compared to the code in the Github repo for this module).

**Comment by lichesser** [comment]
I also just noticed there now is a `docs` folder in the root directory of the installation and a new `config/services/webservice/` folder which contains some `*.yml` files

**Comment by khouloudbelguith** [comment]
@lichesser, 
>config/services/webservice/ folder

This is a normal behavior => here's a PR: https://github.com/PrestaShop/PrestaShop/pull/14953 of adding this folder.
>there now is a docs folder in the root directory

the docs folder exists in PS1.7.6.0 also.
You can download new PrestaShop v1.7.6.0 & check all folders.

>psaddonsconnect module

After fresh install PS1.7.6.0, PS1.7.5.2 & PS1.7.6.1, the `vendor` folder & `composer.json` exist.

>ps_emailsubscription & ps_facetedsearch

In my case, after upgrade from PS1.7.6.0 to PS1.7.6.1 => the `vendor` folder & `composer.json` do not exist.

But after upgrading those modules =>   the `vendor` folder & `composer.json` are created.

>The <version> tag in config.xml for ps_linklist was changed from 3.0.4 to 3.0.3 but the actually installed code files are definitely version 3.0.4 (compared to the code in the Github repo for this module).

Yes, I manage to reproduce the issue
In the config.xml version =3.0.3 but, it is displayed 3.0.4 in the Module manager page

Thanks to check & feedback.

**Comment by khouloudbelguith** [comment]
Hi @lichesser,

Since we had no news from you for more than 30 days, I'll close this ticket. Feel free to open another one if you can give specific details.

Thanks!

**Comment by lichesser** [comment]
I'm quite surprised that this issue was closed. What further info did you want?

I thought it is sufficiently clear that the PrestaShop installation process and later on the update process for Prestashop and it's modules creates many files which are meant only for development environments. Many of those files are undesirable for a production environment.

Only people developing PrestaShop or PrestaShop modules e.g. need the `composer` files. Also some module folders now contain stuff like `.git` folders and `tests` folders after a module update.

All those files are unneeded and potentially dangerous in a production environment.

I hope this gets reevaluated and a point in the release pipeline for PrestaShop and it's modules is identified where all those unwanted files get removed.

**Comment by khouloudbelguith** [comment]
@lichesser, I 'm waiting for your feedback about my last comment for more than 30 days & I haven't a response from you.
You have different points in this issue.

Thanks!

**Comment by lichesser** [comment]
@khouloudbelguith 

Ok.

Your last comment was a confirmation that you managed to reproduce the issues I reported.
You confirmed that the `composer` files are created for you also and confirmed the issue with the `config.xml` file. 

Thus I had nothing to add to the ticket as you confirmed my report.

Instead I was waiting on information if this behavior is indeed expected. Maybe I did not make sufficiently clear what I thought was the issues with these files being created on a production server. To me it was obvious that those files don't belong on a production server and are used by developers only. I (seemingly wrongly) assumed this was implied by creating this ticket and asking about those files.

So should I make a new issue so that this issue can get escalated/evaluated with your devs and release managers?

**Comment by khouloudbelguith** [comment]
@lichesser, I checked with our developer @PierreRambaud 

This is normal behavior because new versions of modules have new dependencies.
So, all the new versions it's normal to have a vendor file.
When you go through a fresh install that comes from git or download, it is already packaged with Composer but in that of the core.

Thanks for your understanding!

**Comment by khouloudbelguith** [comment]
@lichesser,

>module folders now contain stuff like .git folders 

I tried to make an upgrade of the module ps_emailsubscription & ps_facetedsearch, there is no `.git `file created.

I used the command `ls -l `& `ls -a` commands
ps_emailsubscription
```
-rw-rw-rw- 1 www-data          www-data            423 oct.  24 17:40 composer.json
-rw-rw-rw- 1 www-data          www-data            564 oct.  24 17:40 composer.lock
-rwxrwxrwx 1 khouloud.belguith khouloud.belguith   566 oct.  24 17:40 config.xml
-rw-rw-rw- 1 www-data          www-data            603 oct.  24 17:40 CONTRIBUTORS.md
drwxrwxrwx 3 khouloud.belguith khouloud.belguith  4096 oct.   8 15:25 controllers
-rwxrwxrwx 1 khouloud.belguith khouloud.belguith  1287 oct.  24 17:40 index.php
-rwxrwxrwx 1 khouloud.belguith khouloud.belguith  5537 oct.  24 17:40 logo.png
drwxrwxrwx 9 khouloud.belguith khouloud.belguith  4096 oct.   9 12:37 mails
-rwxrwxrwx 1 khouloud.belguith khouloud.belguith 56181 oct.  24 17:40 ps_emailsubscription.php
-rw-rw-rw- 1 www-data          www-data           2108 oct.  24 17:40 Readme.md
drwxrwxrwx 2 khouloud.belguith khouloud.belguith  4096 oct.   8 15:25 translations
drwxrwxrwx 2 khouloud.belguith khouloud.belguith  4096 oct.  24 17:40 upgrade
drwxrwxrwx 3 www-data          www-data           4096 oct.  24 17:40 vendor
-rwxrwxrwx 1 khouloud.belguith khouloud.belguith   418 oct.  24 17:40 verification.php
drwxrwxrwx 3 khouloud.belguith khouloud.belguith  4096 oct.   8 15:25 views
```
ps_facetedsearch
```
-rw-rw-rw- 1 www-data          www-data            423 oct.  24 17:40 composer.json
-rw-rw-rw- 1 www-data          www-data            564 oct.  24 17:40 composer.lock
-rwxrwxrwx 1 khouloud.belguith khouloud.belguith   566 oct.  24 17:40 config.xml
-rw-rw-rw- 1 www-data          www-data            603 oct.  24 17:40 CONTRIBUTORS.md
drwxrwxrwx 3 khouloud.belguith khouloud.belguith  4096 oct.   8 15:25 controllers
-rwxrwxrwx 1 khouloud.belguith khouloud.belguith  1287 oct.  24 17:40 index.php
-rwxrwxrwx 1 khouloud.belguith khouloud.belguith  5537 oct.  24 17:40 logo.png
drwxrwxrwx 9 khouloud.belguith khouloud.belguith  4096 oct.   9 12:37 mails
-rwxrwxrwx 1 khouloud.belguith khouloud.belguith 56181 oct.  24 17:40 ps_emailsubscription.php
-rw-rw-rw- 1 www-data          www-data           2108 oct.  24 17:40 Readme.md
drwxrwxrwx 2 khouloud.belguith khouloud.belguith  4096 oct.   8 15:25 translations
drwxrwxrwx 2 khouloud.belguith khouloud.belguith  4096 oct.  24 17:40 upgrade
drwxrwxrwx 3 www-data          www-data           4096 oct.  24 17:40 vendor
-rwxrwxrwx 1 khouloud.belguith khouloud.belguith   418 oct.  24 17:40 verification.php
drwxrwxrwx 3 khouloud.belguith khouloud.belguith  4096 oct.   8 15:25 views
```

Thanks!
